### PR TITLE
Update run-tox-env.yml to add Python 3.8 support

### PR DESCRIPTION
**Pull request by laike9m**
Azure pipelines has added Python 3.8 support:
microsoft/azure-pipelines-image-generation#1317

### CODE REVIEW

File: run-tox-env.yml [hunk 1]
```diff
@@ -83,17 +83,13 @@ jobs:
...
             ${{ if or(eq(job.key, 'py37'), startsWith(job.key, 'py37-')) }}:
               versionSpec: '3.7'
+            ${{ if or(eq(job.key, 'py38'), startsWith(job.key, 'py38-')) }}:
+              versionSpec: '3.8'
             ${{ if or(eq(job.key, 'pypy'), startsWith(job.key, 'pypy-')) }}:
               versionSpec: 'pypy2'
             ${{ if or(eq(job.key, 'pypy3'), startsWith(job.key, 'pypy3-')) }}:
               versionSpec: 'pypy3'
           architecture: ${{ coalesce(job.value.architecture, 'x64') }}
-    - ${{ if or(eq(job.key, 'py38'), startsWith(job.key, 'py38-')) }}:
```
Comment (gaborbernat) [inline run-tox-env.yml:91]: I think we should move this to 39 now 👍 but still keep.

**Comment by gaborbernat** [comment]
Seems they haven't yet added it; we'll wait for their release to merge:

```
Version spec 3.8 for architecture x64 did not match any version in Agent.ToolsDirectory.
Versions in /opt/hostedtoolcache:

2.7.16 (x64)
3.4.10 (x64)
3.5.7 (x64)
3.6.9 (x64)
3.7.4 (x64)
If this is a Microsoft-hosted agent, check that this image supports side-by-side versions of Python at https://aka.ms/hosted-agent-software.
If this is a self-hosted agent, see how to configure side-by-side Python versions at https://go.microsoft.com/fwlink/?linkid=871498.
```

**Comment by laike9m** [comment]
> Seems they haven't yet added it; we'll wait for their release to merge:
> 
> ```
> Version spec 3.8 for architecture x64 did not match any version in Agent.ToolsDirectory.
> Versions in /opt/hostedtoolcache:
> 
> 2.7.16 (x64)
> 3.4.10 (x64)
> 3.5.7 (x64)
> 3.6.9 (x64)
> 3.7.4 (x64)
> If this is a Microsoft-hosted agent, check that this image supports side-by-side versions of Python at https://aka.ms/hosted-agent-software.
> If this is a self-hosted agent, see how to configure side-by-side Python versions at https://go.microsoft.com/fwlink/?linkid=871498.
> ```

HI Bernát, may I ask where did you find the reference? It seems many people in this issue microsoft/azure-pipelines-image-generation#1317 have set up CI using Python 3.8 successfully, I wonder how they did it.

**Comment by gaborbernat** [comment]
This PRs CI itself fails with the above error. 🤔 That issue is still open, did people really succeed?

**Comment by laike9m** [comment]
😧

**Comment by asottile** [comment]
AP doesn't have 3.8 yet without using an external package (such as deadsnakes) -- I also haven't set up a 3.9a0 package yet for deadsnakes so that isn't quite ready yet either

**Pull request merged**